ProcessList.c

```diff
@@ -553,13 +554,39 @@ void ProcessList_rebuildPanel(ProcessList* this) {
    bool foundFollowed = false;
 
    for (int i = 0; i < processCount; i++) {
-      Process* p = (Process*) Vector_get(this->processes, i);
-
-      if ( (!p->show)
-         || (this->userId != (uid_t) -1 && (p->st_uid != this->userId))
-         || (incFilter && !(String_contains_i(Process_getCommand(p), incFilter)))
-         || (this->pidMatchList && !Hashtable_get(this->pidMatchList, p->tgid)) )
-         continue;
+       Process* p = (Process*) Vector_get(this->processes, i);
+
+       if ( (!p->show)
+               || (this->userId != (uid_t) -1 && (p->st_uid != this->userId))
+               || (incFilter && !(String_contains_i(Process_getCommand(p), incFilter))) ) {
+           continue;
+       } else {
+           if (this->pidMatchList && !Hashtable_get(this->pidMatchList, p->tgid)) {
+               // iterate over parents and check hashtable for entries
+               pid_t ppid = Process_getParentPid(p);
+               bool preExisting;
+               bool nextPreExisting;
+               Process *ppidProcess = ProcessList_getProcess(this, ppid, &preExisting, LinuxProcess_new);
+               Process *nextProcess;
+               bool found = Hashtable_get(this->pidMatchList, ppidProcess->tgid);
+               while (ppidProcess->tgid != 0 && !found) {
+                   ppid = Process_getParentPid(ppidProcess);
+                   nextProcess = ProcessList_getProcess(this, ppid, &nextPreExisting, LinuxProcess_new);
+                   found = Hashtable_get(this->pidMatchList, nextProcess->tgid);
+                   if (!preExisting) {
+                       free(ppidProcess);
+                   }
+                   preExisting = nextPreExisting;
+                   ppidProcess = nextProcess;
+               }
+               if (ppidProcess->tgid == 0 && !Hashtable_get(this->pidMatchList, ppidProcess->tgid)) {
+                   if (!preExisting) {
+                        free(ppidProcess);
+                   }
+                   continue;
+               }
+           }
+       }
```

Comment:
Code indentation and formatting

Reply:
I changed the indentation to 3 and fixed the braces on the one-line if statements.